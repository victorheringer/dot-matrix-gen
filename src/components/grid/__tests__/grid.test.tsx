import { fireEvent, render } from "@testing-library/react";
import { Grid } from "../index";

describe("Grid", () => {
  it("should render correctly", () => {
    const tree = render(
      <Grid
        matrix={[
          [true, false],
          [true, false],
        ]}
        clickCell={jest.fn()}
        pixelSize={10}
      />
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should call clickCell when a cell is cell is clicked", () => {
    const mockClickCell = jest.fn();

    const { getByLabelText } = render(
      <Grid
        matrix={[
          [true, false],
          [true, false],
        ]}
        clickCell={mockClickCell}
        pixelSize={10}
      />
    );

    fireEvent.click(getByLabelText("cell position 0:0"));

    expect(mockClickCell).toBeCalledTimes(1);
    expect(mockClickCell).toBeCalledWith(0, 0, true);
  });
});
