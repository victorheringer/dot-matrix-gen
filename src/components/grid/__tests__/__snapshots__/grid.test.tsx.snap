// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ ccfSbU"
  >
    <table>
      <tbody>
        <tr>
          <td
            aria-label="cell position 0:0"
            class="sc-gsnTZi ftUeuu"
            size="10"
          />
          <td
            aria-label="cell position 0:1"
            class="sc-gsnTZi boLvHC"
            size="10"
          />
        </tr>
        <tr>
          <td
            aria-label="cell position 1:0"
            class="sc-gsnTZi ftUeuu"
            size="10"
          />
          <td
            aria-label="cell position 1:1"
            class="sc-gsnTZi boLvHC"
            size="10"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
